sdks/java/core/src/main/java/org/apache/beam/sdk/io/LocalResourceId.java

```diff
@@ -35,7 +35,8 @@
  */
 class LocalResourceId implements ResourceId {
 
-  private final Path path;
+  private final String pathString;
+  private transient volatile Path cachedPath;
```

Comment:
I guess I could have used a Guava `Supplier` for this, but too much Guava in Beam already.